torch/distributed/elastic/multiprocessing/api.py

```diff
@@ -117,11 +117,10 @@ def from_str(cls, vm: str) -> Union["Std", Dict[int, "Std"]]:
         Any other input raises an exception
         """
 
-        def to_std(v):
-            v = int(v)
-            for s in Std:
-                if s == v:
-                    return s
+        def to_std(v: str) -> Std:
+            s = Std(int(v))
+            if s in Std:
+                return s
```

Comment:
can we raise a error in the else statement to silent mypy?

Reply:
Could do.. hmm, thinking about it.  The intent here is not to change the behaviour of the code in any way.

Since the `else` should never be reached, then raising an error (like a `ValueError`) _should_ be fine.  In this case it might be preferable to use a `# type: ignore[return]` comment.

Reply:
Also: was `mypy` not complaining about that before these changes?  That seems like a possible bug.

Reply:
Eugh, perhaps it's more complicated.  At a guess: maybe `mypy` was able to infer from the `for` loop that all values are checked, and it doesn't do the same for the membership test.

Reply:
:facepalm: Nope, a much simpler explanation :)

`mypy` did not complain previously because there were no type hints on the method.

d19f8f2f7385d8f501b481c5bcff15361e963f15 introduced type hints, and then the lint error appeared.